src/components/core/PivotTable.tsx

```diff
@@ -414,6 +420,51 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
         }
     };
 
+    private startWatchingTableRendered = () => {
+        // table having no data will be unmounted, it causes ref null
+        if (!this.containerRef) {
+            this.stopWatchingTableRendered();
+            return;
+        }
+
+        // table has data and takes place of Loading icon
+        if (!this.isTableHidden()) {
+            this.stopWatchingTableRendered();
+            setTimeout(this.props.afterRender, 100);
+        }
+    };
+
+    private stopWatchingTableRendered = () => {
+        clearInterval(this.watchingIntervalId);
+        this.watchingIntervalId = null;
```

Comment:
should we clear `watchingTimeoutId` also?

Reply:
I will be cleaned in `watchingTableRenderedTimeout`

Reply:
In this case, after `afterRender` is called, `watchingTableRenderedTimeout` will be called also when `WATCHING_TABLE_RENDERED_MAX_TIME` is met.

Reply:
Fixed.
You're right, we should clean timeout ASAP.